### PR TITLE
Sing up test

### PR DESCRIPTION
**Pull request by jupabass89**
## Summary
-The users controller test were implemented and tested succesfully.

## Trello Card
https://trello.com/c/oQYgvHWn/27-tests-de-sign-up

### CODE REVIEW

File: test/app/controllers/users.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const request = require('supertest');
+const app = require('../../../');
+const { mookUser } = require('../../mooks/users');
```
Comment (arinaldi118) [inline test/app/controllers/users.test.js]: mook?? :thinking: I think you meant `mock`

File: app/middlewares/errors.js [hunk 2]
```diff
@@ -4,11 +4,11 @@ const logger = require('../logger');
 const DEFAULT_STATUS_CODE = 500;
 
 const statusCodes = {
-  [errors.DATABASE_ERROR]: 503,
+  [errors.DATABASE_ERROR]: 500,
```
Comment (arinaldi118) [inline app/middlewares/errors.js]: database error is 503

File: app/controllers/users.js [hunk 3]
```diff
@@ -9,10 +9,10 @@ exports.registerUser = (req, res, next) => {
   return userService
     .validateUser(user)
     .then(responseUser => {
-      if (responseUser !== null) return Promise.reject(errors.databaseError('user already exists'));
+      if (responseUser) return Promise.reject(errors.invalidData('user already exists'));
```
Comment (arinaldi118) [inline app/controllers/users.js]: if user exists you must return a different error. For example, an error with status code 409

File: app/models/usuario.js [hunk 4]
```diff
@@ -9,7 +9,8 @@ module.exports = (sequelize, DataTypes) => {
       password: { type: DataTypes.STRING, allowNull: false }
     },
     {
-      tableName: 'users'
+      tableName: 'users',
+      underscored: true
     }
   );
   return User;
```
Comment (arinaldi118) [inline app/models/usuario.js]: Change the name of this file

File: app/middlewares/errors.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ const statusCodes = {
   [errors.DATABASE_ERROR]: 503,
   [errors.DEFAULT_ERROR]: 500,
   [errors.REQUEST_ERROR]: 500,
-  [errors.EMPTY_ERROR]: 404,
+  [errors.USER_EXISTS_ERROR]: 409,
   [errors.INVALID_DATA_ERROR]: 404
```
Comment (arinaldi118) [inline app/middlewares/errors.js]: schema error is 422

File: test/app/controllers/users.test.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+          .post('/users')
+          .send(user)
+          .expect(422)
+      ));
+
+  it('should fail if a user insert a bad password', () =>
+    request(app)
+      .post('/users')
+      .send({ ...user, password: '123' })
+      .expect(404));
```
Comment (arinaldi118) [inline test/app/controllers/users.test.js]: 422

File: test/app/controllers/users.test.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  it('should fail due to null parameters', () => {
+    delete user.name;
+    delete user.password;
+    return request(app)
+      .post('/users')
+      .send(user)
+      .expect(404);
```
Comment (arinaldi118) [inline test/app/controllers/users.test.js]: 422

File: test/app/controllers/users.test.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+  it('should fail if a email already exist', () =>
+    request(app)
+      .post('/users')
+      .send(user)
+      .then(
+        request(app)
+          .post('/users')
+          .send(user)
+          .expect(422)
```
Comment (arinaldi118) [inline test/app/controllers/users.test.js]: 409

File: app/models/usuario.js (hunk 4 above)
Comment (jupabass89) [reply app/models/usuario.js]: I've changed this file name but git does not seem to recognize that this change is in another file.

**Pull request merged**